File: ruby/lib/coin.rb
class Coin
  attr_reader :coins

  DEFAULT_COINS = [50, 20, 10, 5, 2, 1]

  def initialize
    @coins = DEFAULT_COINS
  end

  def change(amount)
    result = []
    @coins.each do |c|
      if amount >= c
        count = (amount / c).floor
        result.concat([c] * count)
        amount -= (c * count)
      end
    end
    result
  end

end

File: javascript/src/coin.js
'use strict';

function Coin(DEFAULTCOIN = [50, 20, 10, 5, 2, 1]){
  this.coins = DEFAULTCOIN;
};

Coin.prototype.change = function (amount) {
  const result = []
  this.coins.forEach(function(c) {
    if (amount >= c) {
      var count = Math.floor(amount / c)
      for(var i = 0; i < count; i++){
        result.push(c);
      }
      amount -= (c * count);
    }
  });
  return result;
};
